packaging/nnstreamer.spec

```diff
@@ -411,7 +414,8 @@ Static library package of nnstreamer-devel.
 %package test-devel
 Summary: Development package to provide testable environment of a subplugin (tensor_filter/custom)
 Requires: nnstreamer-devel = %{version}-%{release}
-Conflicts: nnstreamer-configuration
+Provides: nnstreamer-configuration = %{version}-%{release}
```

Comment:
basic question - where is `nnstreamer-configuration` package defined? 
Also, both `default-configuration` and `test-devel` packages `conflicts` each other but in the end are providing the same package - `nnstreamer-configuration`. what am I missing here?

Reply:
This is called virtual package.

So, `test-devel` and `default-configuration` provides the capability of `nnstreamer-configuration` which is virtual capability 

When installing nnstreamer, it is `recommended` to have `default-configuration` to satisfy the capability of `nnstreamer-configuration` but even if our system has `test-devel`, nnstreamer will be happy to accept it as a `nnstreamer-configuration`, so this makes the other packager to choose which one to include at will.

So, when we need to have test environment, we require `nnstreamer-test-devel`. The dependency solver will be happy to have `nnstreamer-test-devel` instead of `nnstreamer-default-configuration` to satisfy `nnstreamer-configuration`

referring to below.
https://docs.fedoraproject.org/en-US/Fedora_Draft_Documentation/0.1/html/RPM_Guide/ch-advanced-packaging.html#id818305

https://fedoraproject.org/wiki/Packaging:WeakDependencies#Real_life_example

Reply:
Got it. Thanks for the explanation.

Reply:
It might be helpful to add a note that `nnstreamer-configuration` is a virtual capability.

Reply:
Updated :)